x-pack/plugins/security_solution/public/timelines/components/flyout/pane/pane.tsx

```diff
@@ -5,16 +5,18 @@
  * 2.0.
  */
 
-import React, { useMemo, useRef } from 'react';
-import { css } from '@emotion/react';
-import { useEuiBackgroundColor, useEuiTheme } from '@elastic/eui';
-
+import React, { useRef } from 'react';
+import classNames from 'classnames';
 import { StatefulTimeline } from '../../timeline';
 import type { TimelineId } from '../../../../../common/types/timeline';
 import * as i18n from './translations';
 import { defaultRowRenderers } from '../../timeline/body/renderers';
 import { DefaultCellRenderer } from '../../timeline/cell_rendering/default_cell_renderer';
 import { EuiPortal } from './custom_portal';
+import { useShallowEqualSelector } from '../../../../common/hooks/use_selector';
+import { inputsSelectors } from '../../../../common/store/selectors';
+import { usePaneStyles, OverflowHiddenGlobalStyles } from './pane.styles';
+
 interface FlyoutPaneComponentProps {
   timelineId: TimelineId;
   visible?: boolean;
```

Comment:
This all looks good to me, just adding that I think we should use the property isTimelineBottomBarVisible from x-pack/plugins/security_solution/public/app/home/template_wrapper/index.tsx in conjunction with the show property that comes from redux in x-pack/plugins/security_solution/public/timelines/components/flyout/index.tsx to add this display: none; css, and then remove the conditional rendering of the timeline in the template wrapper component. This would result in the timeline mounting only once when the user enters the app, instead of every time they go to a menu/management page, cloud security, etc. Depending on what routes a user takes through the app, this could significantly improve performance of the app, especially if timeline contains a query result of some sort at the time.

Reply:
@kqualters-elastic Yeah, that would have a good impact in some cases, because we would not need to re-fetch the query and re-render when navigating from a "timeline page" to a "non-timeline page" and then to a "timeline page" again, we would do it only once, right now we are fetching and rendering twice in this scenario. 

However, this would also have a bad impact if we don't need to use the timeline, because if the user lands on a "non-timeline" page we would be always fetching the query and loading all the timeline components in the background, even if the user never uses the timeline.
I am not sure we should assume the user will always need the timeline loaded and ready, even on pages where the timeline is disabled. If the user never navigates to a "timeline page" it is better to not mount it at all.

I don't have a strong opinion about it, I can see good and bad things, and maybe it is worth it. I think we should bring this proposal to Paul 👍

Reply:
@semd , was there discussion and final decision on this? As I can see currently, even without user action, timeline modal is loaded in the background as soon as the user enters the page..

Reply:
@logeekal What you described is the current behavior, the Timeline is loaded and rendered in the background when a page with Timeline **enabled** is loaded. Kevin is suggesting treating pages with Timeline enabled and disabled the same way, just hiding the Timeline on pages with Timeline disabled, instead of unmounting it. No decision has been taken yet.

Regarding the approach you and I had talked about, to not load the timeline until the user opens it explicitly. I implemented the behavior, but it caused a bug on the count badge that is showing in the Timeline bottom bar for saved timelines, it was not displayed after a fresh load, it was showing only after the Timeline was open, obviously 😄. So I decided to roll back everything, keeping the current behavior.

Probably the Timeline loading/rendering strategy can be improved, we can talk about it, but I think that if we make some changes on that matter we can do it in a separate PR, and keep this one as visual changes only.

Reply:
I agree. thank you explaining. Did not think initially about the count. Other than that. PR looks great 🚀

Reply:
Me neither... Okay. Thanks 👍